client/lib/media/utils.js

```diff
@@ -535,6 +535,14 @@ const MediaUtils = {
 				extension: MediaUtils.getFileExtension( file ),
 				mime_type: MediaUtils.getMimeType( file )
 			} );
+		} else if ( file.thumbnails ) {
```

Comment:
Use of this was actually in another PR, so I've removed it from this PR and I'll include it there -
 it's a more suitable place to show the whole flow.